.github/workflows/packer-build-and-publish.yml

```diff
@@ -0,0 +1,223 @@
+        id: build
+        env:
+          GITHUB_REF: ${{ github.ref }}
+          GITHUB_JOB_WORKFLOW_REF: ${{ steps.auth.outputs.job_workflow_ref }}
+          GITHUB_REPOSITORY_NAME: ${{ steps.auth.outputs.repository_name }}
+        run: |
+          packer init ${{ matrix.template }}
+          packer build \
+            -var provider=aws \
+            -var aws_private_ami=true \
+            -var image_version="${{ github.ref_name }}" \
+            -var aws_assume_role_arn="arn:aws:iam::${{ matrix.account }}:role/github-actions/packer-role" \
+            -var aws_assume_role_session_name=GitHubActions \
+            ${{ matrix.template }}
+
+          echo "name=$(basename ${{ matrix.template }})" >> "${GITHUB_OUTPUT}"
+          echo "readme=$(git diff --name-only -- '*.md')" >> "${GITHUB_OUTPUT}"
+          echo "report=$(git ls-files --others --exclude-standard -- '*.json')" >> "${GITHUB_OUTPUT}"
+
+      - name: Upload software report
+        uses: actions/upload-artifact@834a144ee995460fba8ed112a2fc961b36a5ec5a # v4.3.6
+        with:
+          name: "Software Report for ${{ steps.build.outputs.name }}"
+          path: ${{ steps.build.outputs.report }}
+          overwrite: true
+
+      - name: Upload readme
+        uses: actions/upload-artifact@834a144ee995460fba8ed112a2fc961b36a5ec5a # v4.3.6
+        with:
+          name: "Readme for ${{ steps.build.outputs.name }}"
+          path: ${{ steps.build.outputs.readme }}
+          overwrite: true
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Download artifacts
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          path: artifacts
+
+      - name: Update readmes
+        run: |
+          # Define the base directories
+          artifacts_dir="artifacts"
+          images_dir="images"
+
+          # Loop through all .pkr.hcl files in the images directory
+          find "$images_dir" -type f -name "*.pkr.hcl" | while read -r pkr_file; do
+              # Extract the template filename without the path and extension
+              pkr_filename=$(basename "$pkr_file")
+              
+              # Extract the possible readme folder name based on the .pkr.hcl file name
+              readme_folder="Readme for ${pkr_filename}/"
+              
+              # Find the corresponding readme file within that folder
+              readme_file=$(find "$artifacts_dir" -type f -path "*/${readme_folder}*" -name "*.md")
+              
+              # If the readme file exists, move it to the directory containing the templates folder
+              if [[ -f "$readme_file" ]]; then
+                  destination_dir=$(dirname "$pkr_file")/../
+                  mv -f "$readme_file" "$destination_dir"
+              fi
+          done
+
+      - name: Update reports
```

Comment:
```suggestion
      - name: Move reports
```

Reply:
If we want't to make this more precise we can change it to "Rename reports", we don't actually move them anywhere, just rename the file.
```
              current_file="$report_dir/software-report.json"
              new_file="$report_dir/software-report-${pkr_name}.json"
```

Reply:
Done in https://github.com/grafana/runner-images/pull/111/commits/e0212ceb5ab467a86ea8cf2a56c5c21179bf19bb